packages/components/src/validated-form-controls/control-with-error.tsx

```diff
@@ -0,0 +1,161 @@
+import { Icon } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { error } from '@wordpress/icons';
+import { cloneElement, forwardRef, useEffect, useState } from 'react';
+import { withIgnoreIMEEvents } from '../utils/with-ignore-ime-events';
+
+import './style.scss';
+
+function appendRequiredIndicator(
+	label: string,
```

Comment:
This is fine, but do we need to support the label to be a React node?

It's allowed to be React node in `InputBaseProps`.

https://github.com/WordPress/gutenberg/blob/accde16096464ac832fbe8e249a3b61535800800/packages/components/src/input-control/types.ts#L179

Reply:
Good catch, some `label` props accept `string` and some accept `ReactNode`. Updated to support both cases in 668e8444885a838ffc67ea33d75d3e38f6f72f24.